src/Materials/shaderMaterial.ts

```diff
@@ -147,9 +147,10 @@ export class ShaderMaterial extends Material {
      *  * object: { vertexSource: "vertex shader code string", fragmentSource: "fragment shader code string" } using with strings containing the shaders code
      *  * string: "./COMMON_NAME", used with external files COMMON_NAME.vertex.fx and COMMON_NAME.fragment.fx in index.html folder.
      * @param options Define the options used to create the shader
+     * @param storeEffectOnSubMeshes true to store effect on submeshes, false to store the effect directly in the material class.
      */
-    constructor(name: string, scene: Scene, shaderPath: any, options: Partial<IShaderMaterialOptions> = {}) {
-        super(name, scene);
+    constructor(name: string, scene: Scene, shaderPath: any, options: Partial<IShaderMaterialOptions> = {}, storeEffectOnSubMeshes = true) {

```

Comment:
Would it work if storeEffectOnSubMeshes is false with a pushMaterial ??? I always thought a push mat would need to store the effect on a submesh ?

Reply:
Yes, a `ShaderMaterial` can work in both mode (that's the point of the PR actually), that's why it's a parameter of the constructor. The other materials (standard, PBR, etc) would probably not work but you can't change the value (`_storeEffectOnSubMeshes` is always `true`) as it is not exposed.